File: C#/Eliminacion_selec.cs
using System;
using System.Collections.Generic;

class Program {
    static void Main() {
        //uso una lista para poder eliminar elementos del array
        List<int> eliminacionselec = new List<int> { 11, 21, 31, 41, 51, 61 };
        int selec = 3;
        int largo = eliminacionselec.Count; //variable que guarda el largo del arreglo

        //Recorro la lista para mostrar el arreglo
        Console.Write("Antes de la eliminacion del elemento en el indice seleccionado, el array es: ");
        for (int indice = 0; indice < largo; indice++) {
            Console.Write(eliminacionselec[indice] + " ");
        }
        Console.WriteLine();

        //Elimina el elemento del indice seleccionado (-1 porque los array empiezan en 0)
        eliminacionselec.RemoveAt(selec -1);
        
        //recorre el arreglo con la eliminacion hecha
        Console.Write("Después de la eliminacion del elemento en el indice seleccionado, el array es: ");
        for (int indice = 0; indice < largo; indice++) {
            Console.Write(eliminacionselec[indice] + " ");
        }
        Console.WriteLine();
    } 
}

File: C++/Eliminacion_selec.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    //uso vector para poder insertar elementos a un arreglo
    vector<int> eliminaciones = {11, 21, 31, 41, 51, 61};
    int largo = eliminaciones.size();
    int idx = 3; //indice del elemento que se va a eliminar

    //recorre el array antes de eliminar nada
    cout << "Antes de la eliminacion, el array es: ";
    for (int indice = 0; indice < largo; indice++) {
        cout << eliminaciones[indice] << " ";
    }
    cout << endl;
    
    //elimino el elemento contando las posiciones desde el inicio del arreglo
    eliminaciones.erase(eliminaciones.begin() + (idx - 1));
    
    cout << "Después de la eliminacion selectiva del array es: ";
    for (int indice = 0; indice < largo; indice++) {
        cout << eliminaciones[indice] << " ";
    }
    cout << endl;
    
    return 0;
}